src/TLuaInterpreter.cpp

```diff
@@ -11098,58 +11103,78 @@ int TLuaInterpreter::addSupportedTelnetOption( lua_State *L )
     return 0;
 }
 
+// Although this is coded here as "Echo" it is registered in the Lua system as
+// "echo" - so THAT is the name that should be displayed as the function name!
 int TLuaInterpreter::Echo( lua_State *L )
 {
-    string a1;
-    string a2;
-    int s = 1;
-    int n = lua_gettop( L );
-
-    if( ! lua_isstring( L, s ) )
-    {
-        lua_pushstring( L, "Echo: wrong argument type" );
-        lua_error( L );
-        return 1;
-    }
-    else
-    {
-        a1 = lua_tostring( L, s );
-        s++;
-    }
-    if( n > 1 )
-    {
-        if( ! lua_isstring( L, s ) )
-        {
-            lua_pushstring( L, "Echo: wrong argument type" );
-            lua_error( L );
-            return 1;
-        }
-        else
-        {
-            a2 = lua_tostring( L, s );
-        }
-    }
     Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
-    QString txt;
-    QString name;
-    if( n == 1 )
-    {
-        txt = a1.c_str();
-        pHost->mpConsole->buffer.mEchoText = true;
-        pHost->mpConsole->echo( txt );
-        pHost->mpConsole->buffer.mEchoText = false;
+    if (! pHost) {
+        lua_pushstring(L, tr("echo: NULL Host pointer - something is wrong!")
+                       .toUtf8().constData());
+        return lua_error(L);
     }
 
+    QString consoleName;
+    QString displayText;
+    int n = lua_gettop(L);
```

Comment:
:warning: Humm, if `n` > 2 it will use the _last_ argument as the text to echo, rather than the 2nd, a bit of a corner case but perhaps I should've limited n to be <= 2...?